ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/LondonTargetingGasLimitCalculator.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.mainnet;
+
+import org.hyperledger.besu.ethereum.mainnet.feemarket.LondonFeeMarket;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class LondonTargetingGasLimitCalculator extends FrontierTargetingGasLimitCalculator {
+  private static final Logger LOG = LogManager.getLogger();
+  private final long londonForkBlock;
+
+  public LondonTargetingGasLimitCalculator(final long londonForkBlock) {
+    this(1024L, 5000L, Long.MAX_VALUE, londonForkBlock);
+  }
+
+  public LondonTargetingGasLimitCalculator(
+      final long maxConstantAdjustmentIncrement,
+      final long minGasLimit,
+      final long maxGasLimit,
+      final long londonForkBlock) {
+    super(maxConstantAdjustmentIncrement, minGasLimit, maxGasLimit);
+    this.londonForkBlock = londonForkBlock;
+  }
+
+  @Override
+  public long nextGasLimit(
+      final long currentGasLimit, final long targetGasLimit, final long newBlockNumber) {
+    long nextGasLimit;
+    if (targetGasLimit > currentGasLimit) {
+      nextGasLimit = Math.min(targetGasLimit, safeAddAtMost(currentGasLimit));
+    } else if (targetGasLimit < currentGasLimit) {
+      nextGasLimit = Math.max(targetGasLimit, safeSubAtMost(currentGasLimit));
+    } else {
+      nextGasLimit = currentGasLimit;
+    }
+
+    if (londonForkBlock == newBlockNumber) {
+      nextGasLimit = nextGasLimit * LondonFeeMarket.DEFAULT_SLACK_COEFFICIENT;
```

Comment:
I wonder if it would be better if we pass the correct FeeMarket when defining the protocol schedule (passing feemarket in the constructor https://github.com/hyperledger/besu/pull/2632/files#diff-22b78733e37a697fa8d1d8a02d2a87fe5ccea9cf67c34ce5e6311f024c14abd6R480). Because if the DEFAULT_SLACK_COEFFICIENT may change in a future fork that would force us to duplicate that LondonTargetingGasLimitCalculator class just to change that call `LondonFeeMarket.DEFAULT_SLACK_COEFFICIENT` to ` NewForkFeeMarket.DEFAULT_SLACK_COEFFICIENT`. If we pass the fee market  in the constructor the problem is fixed

Reply:
Sure I think that is reasonable